python/sglang/srt/layers/torchao_utils.py

```diff
@@ -47,6 +47,32 @@ def filter_fn(module, fqn):
             256,
         ], f"int4wo groupsize needs to be one of [32, 64, 128, 256] but got {group_size}"
         quantize_(model, int4_weight_only(group_size=group_size), filter_fn=filter_fn)
+    elif "gemlite" in torchao_config:
+        # gemlite-<packing_bitwidth>-<bit_width>-<group_size> or
+        # gemlite-<bit_width>-<group_size> (packing_bitwidth defaults to 32)
+        import os, pwd
+        import gemlite
+        from gemlite.core import GemLiteLinearTriton, set_autotune
+        from torchao.quantization import gemlite_uintx_weight_only
+
+        _quant_args = torchao_config.split("-")
+        bit_width = int(_quant_args[-2])
+        group_size = None if _quant_args[-1] == 'None' else int(_quant_args[-1])
+        try:
+            packing_bitwidth = int(_quant_args[-3])
+        except:
+            # if only 2 inputs found, use default value
+            packing_bitwidth = 32
+
+        quantize_(model, gemlite_uintx_weight_only(group_size, bit_width, packing_bitwidth))
+
+        # try to load gemlite kernel config
+        try:
+            GemLiteLinearTriton.load_config(f"/tmp/{pwd.getpwuid(os.getuid()).pw_gecos}_gemlite.json")
+            print(f"loaded gemlite kernel cache /tmp/{pwd.getpwuid(os.getuid()).pw_gecos}_gemlite.json")
+        except:
```

Comment:
cc @mobicham why do we need this try except? seems that GemLiteLinearTriton is already swallowing errors: https://github.com/mobiusml/gemlite/blob/5ebcca036943e7b1929f9be4beeeb0cf8d3fd170/gemlite/core.py#L427 (it probably should not do that I feel)